### DISCUSSION TITLE
exercise unit 1

### DISCUSSION THREAD

**Discussion by 007ekho** [Lightning-AI/dl-fundamentals#58] [Exercises]
any guide on early stopping?

**Comment by rasbt** [comment]
Sure! As a little hint, you have the `error_count` variable there, which tracks the number of errors in a given epoch (pass over the training set). So, if the `error_count` is 0, there is no need to continue the training for more epochs ...

(Please let me know if this helps or whether you need an additional tip)

**Comment by BiffoQ** [comment]
> any guide on early stopping?

I used this:

if error_count == 0:
            print('perceptron automatically stops')
            break

and it worked.

**Reply by rasbt** [reply]
Yes, this looks correct to me!
